### PR TITLE
[NEXUS-4996] Allow mix case in email top level domain

### PR DESCRIPTION
**Pull request by adreghiciu**
Signed-off-by: Alin Dreghiciu adreghiciu@gmail.com

### CODE REVIEW

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/rest/global/SmtpSettingsValidationPlexusResource.java [hunk 1]
```diff
@@ -131,4 +128,13 @@ public Object put( Context context, Request request, Response response, Object p
         return null;
     }
 
+    static void validateEmail( final String email )
+        throws ResourceException
+    {
+        if ( !EMAIL_PATTERN.matcher( email ).matches() )
```
Comment (peterlynch) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/rest/global/SmtpSettingsValidationPlexusResource.java:139]: null email will fail here. lets prevent potential NPE, even though theory is UI will never pass it.

Comment (adreghiciu) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/rest/global/SmtpSettingsValidationPlexusResource.java:139]: I just extracted the method so it is testable. I could check for null, without any problem.

**Comment by nabcos** [comment]
+1

**Comment by adreghiciu** [comment]
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/350/

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/rest/global/SmtpSettingsValidationPlexusResource.java [hunk 2]
```diff
@@ -131,4 +129,17 @@ public Object put( Context context, Request request, Response response, Object p
         return null;
     }
 
+    static void validateEmail( final String email )
```
Comment (cstamas) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/rest/global/SmtpSettingsValidationPlexusResource.java:132]: I would move this method out of REST layer (into NexusEMailer?) to make it reusable.... it was totally wrong to have it here in 1st place

Comment (adreghiciu) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/rest/global/SmtpSettingsValidationPlexusResource.java:132]: NexusEmailer is an iface. Maybe to some util class in same package?

**Comment by adreghiciu** [comment]
CI voted +1 (The failures were unrelated)

**Comment by cstamas** [comment]
+1 merge this for now, and will move stuff out to emailer, as it is pending release anyway

**Pull request merged**